libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1447,11 +1447,21 @@ void ExpressionCompiler::appendOrdinaryBinaryOperatorCode(Token::Value _operator
 
 void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Type const& _type)
 {
-	IntegerType const& type = dynamic_cast<IntegerType const&>(_type);
-	bool const c_isSigned = type.isSigned();
+//	IntegerType const& type = dynamic_cast<IntegerType const&>(_type);
+	auto const* intType = dynamic_cast<IntegerType const*>(&_type);
+	auto const* fixedType = dynamic_cast<FixedPointType const*>(&_type);
 
-	if (_type.category() == Type::Category::FixedPoint)
-		solUnimplemented("Not yet implemented - FixedPointType.");
+	u256 fixedFactor = fixedType ? pow(u256(10), fixedType->fractionalDigits()) : u256(0);
+
+	bool isSigned;
+
+	if (intType)
```

Comment:
This could be shortened to `bool isSigned = intType ? intType->isSigned() : fixedType->isSigned();` :)